### PR TITLE
Simplify and speed up converter testing

### PR DESCRIPTION
**Pull request by mvdbeek**
(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: .github/workflows/converter_tests.yaml [hunk 1]
```diff
@@ -59,24 +59,13 @@ jobs:
...
-            json=$(mktemp -u -p json_output --suff .json)
-            planemo test --test_output_json "$json" --galaxy_python_version ${{ matrix.python-version }} --galaxy_root 'galaxy root' "$CONVERTER" || EXIT_CODE=$?
-          done
-          planemo merge_test_reports json_output/*.json tool_test_output.json
-          planemo test_reports tool_test_output.json --test_output tool_test_output.html
-          if [ "$EXIT_CODE" -ne 0 ]; then
-            echo "Unsuccessful tests found, inspect the 'Converter test results' artifact for details."
-            exit $EXIT_CODE
-          fi
+          planemo test --galaxy_python_version ${{ matrix.python-version }} --galaxy_root 'galaxy root' "$CONVERTER" --test_data 'galaxy root/test-data/' --no_conda_auto_init --biocontainers "${TOOL_ARRAY[@]}"
```
Comment (nsoranzo) [inline .github/workflows/converter_tests.yaml]:
```suggestion
          planemo test --galaxy_python_version ${{ matrix.python-version }} --galaxy_root 'galaxy root' --test_data 'galaxy root/test-data/' --no_conda_auto_init --biocontainers "${TOOL_ARRAY[@]}"
```

Comment (mvdbeek) [reply .github/workflows/converter_tests.yaml]: Nice, that was the missing piece.

**Comment by mvdbeek** [comment]
Looks like bigwig outputs failed ... because the len files were not mounted into the container ?!

File: .github/workflows/converter_tests.yaml [hunk 2]
```diff
@@ -59,24 +59,11 @@ jobs:
...
-            json=$(mktemp -u -p json_output --suff .json)
-            planemo test --test_output_json "$json" --galaxy_python_version ${{ matrix.python-version }} --galaxy_root 'galaxy root' "$CONVERTER" || EXIT_CODE=$?
-          done
-          planemo merge_test_reports json_output/*.json tool_test_output.json
-          planemo test_reports tool_test_output.json --test_output tool_test_output.html
-          if [ "$EXIT_CODE" -ne 0 ]; then
-            echo "Unsuccessful tests found, inspect the 'Converter test results' artifact for details."
-            exit $EXIT_CODE
-          fi
+          planemo test --galaxy_python_version ${{ matrix.python-version }} --galaxy_root 'galaxy root' --test_data 'galaxy root/test-data/' "${TOOL_ARRAY[@]}"
```
Comment (nsoranzo) [inline .github/workflows/converter_tests.yaml]: Looks like we don't need to set `--test_data` for https://github.com/galaxyproject/galaxy/pull/13909 ? 

```suggestion
          planemo test --galaxy_python_version ${{ matrix.python-version }} --galaxy_root 'galaxy root' "${TOOL_ARRAY[@]}"
```

Comment (mvdbeek) [reply .github/workflows/converter_tests.yaml]: True, but IMO this is the right thing to do so we don't depend on the test data resolver hack.

Comment (nsoranzo) [reply .github/workflows/converter_tests.yaml]: I think it's a good test that the hack still works, until we eventually decide to remove it.

**Comment by mvdbeek** [comment]
Do we want to do biocontainers testing or conda ? Both should work now.

**Comment by nsoranzo** [comment]
> Do we want to do biocontainers testing or conda ? Both should work now.

No strong preference from me.

**Comment by mvdbeek** [comment]
I guess there's something to be said for some conda testing, given the "environment rot". This is ready from my side.

**Pull request merged**